README.md

```diff
@@ -1,2 +1,331 @@
+
+<!-- Diagram covering the major architectural concepts  -->
+
+## 2.1 Accounts
+
+An Farcaster account is similar to an account on pseudonymous social networks like Twitter or Reddit. An individual can operate several accounts as the same time, like a real-name account, a pseudonymous account and a company account.
+
+An account is created by minting a new account number from the Account contract on-chain. Each account number is like a primary key in a database, and points to a user-controlled address known as the `custody address`. Once an account is minted, the custody address can sign messages on behalf of the account. Accounts can be enriched by adding a profile picture, display name, biography and verified usernames like `alice.eth`, which are all done off-chain with signed messages. 
+
+## 2.2 Signed Messages
+
+Signed Messages are cryptographically signed messages from an account that are **tamper-proof** and **self-authenticating**. 
+
+A Signed Message has a **message** object, which contains the data that will be signed. The message is then serialized, and then hashed and signed with the custody address. The hash, signature and signer public key are used to construct the **envelope**, which proves that the message is authentic and has not been modified. Any observer can verify the message by checking that: 
+
+1. The account number is owned by an address whose public key matches the public key in the envelope
+2. The serialized hash of message matches the hash in the envelope
+3. The signature in the envelope is verified by the signing scheme with the hash and public key.   
+
+Messages must contain an account number, to look up the `signerPubKey` on-chain, and a `timestamp`, to order messages. Timestamps are client provided, unverified and should be considered best-effort, since they are vunlerable to [clock skew](https://en.wikipedia.org/wiki/Clock_skew) and [clock drift](https://en.wikipedia.org/wiki/Clock_drift). Users who want to ensure perfect ordering can use [hybrid clocks](https://martinfowler.com/articles/patterns-of-distributed-systems/hybrid-clock.html) to generate timestamps.
+
+```ts
+type SignedMessage = {
+  message: {
+    body: any; 
+    account: number;
+    timestamp: number;
+  };
+  envelope: {
+    hash: string;
+    hashType: 'BLAKE2b';
+    signature: string;
+    signatureType: 'ed25519' | 'ecdsa-secp256k1';
+    signerPubKey: string;
+  }
+};
+```
+
+Object serialization must be done according to [RFC-8785](https://datatracker.ietf.org/doc/html/rfc8785), hashing must be performed with [BLAKE2b](https://www.rfc-editor.org/rfc/rfc7693.txt) and signing must be performed with an Ed25519 signature scheme, or in rare cases an ECDSA secp256k1 scheme. 
```

Comment:
"or in rare cases an ECDSA secp256k1 scheme. " Is this worth specifying in the overview? Or can this detail be pushed to a more in-depth part of the documentation?

Reply:
yeah, good point, resolving this for now.